maas/data_source_maas_machine.go

```diff
@@ -0,0 +1,82 @@
+package maas
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/maas/gomaasclient/client"
+)
+
+func dataSourceMaasMachine() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceMachineRead,
+
+		Schema: map[string]*schema.Schema{
+			"power_type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"power_parameters": {
```

Comment:
This one will not work for machines with IPMI `power_parameters` that include `workaround_flags`. A similar approach to the `resource_maas_machine` should be followed here.

Reply:
Good catch!